azure/services/vmextensions/client.go

```diff
@@ -60,20 +60,12 @@ func (ac *azureClient) Get(ctx context.Context, resourceGroupName, vmName, name
 	return ac.vmextensions.Get(ctx, resourceGroupName, vmName, name, "")
 }
 
-// CreateOrUpdate creates or updates the virtual machine extension
-func (ac *azureClient) CreateOrUpdate(ctx context.Context, resourceGroupName, vmName, name string, parameters compute.VirtualMachineExtension) error {
+// CreateOrUpdateAsync creates or updates the virtual machine extension.
+func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, resourceGroupName, vmName, name string, parameters compute.VirtualMachineExtension) error {
 	ctx, span := tele.Tracer().Start(ctx, "vmextensions.AzureClient.CreateOrUpdate")
 	defer span.End()
 
-	future, err := ac.vmextensions.CreateOrUpdate(ctx, resourceGroupName, vmName, name, parameters)
-	if err != nil {
-		return err
-	}
-	err = future.WaitForCompletionRef(ctx, ac.vmextensions.Client)
-	if err != nil {
-		return err
-	}
-	_, err = future.Result(ac.vmextensions)
+	_, err := ac.vmextensions.CreateOrUpdate(ctx, resourceGroupName, vmName, name, parameters)
```

Comment:
@devigned thoughts on this approach of doing "async" reconciliation? Is it significantly less efficient than storing the future or since we only care about eventually getting the extension status do you think it's appropriate in this situation?

Reply:
I'm thinking in the future we can move this over to VM CreateOrUpdate once that's async. For now the extension could be a long-running operation (timeout is 20 minutes) so we don't want that to be part of a blocking operation

Reply:
I think this should be fine. Though, it would be easy to call this multiple times if one is not tracking an ongoing operation via a secret. With the AMP future, we check to see if the operation is done prior to proceeding, which the Azure SDK for Go can help with. In this case, you'd be responsible for managing that state.